contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+            I3Pool(_3pool).add_liquidity(
+              _add3poolLiquidityAmounts,
+              0 // minimum 0 3crv lp tokens out (defensively checked below)
+            );
+        }
+
+        // get the number of 3crv held by the contract, and also
+        // check for slippage during the 3pool deposit
+        uint256 _3crvBalanceAfter = IERC20(_3crv).balanceOf(address(this));
+        uint256 _3crvBalanceFromDeposit = _3crvBalanceAfter - _3crvBalanceBefore;
+        uint256 _min3crvOut = (_daiBalance + _usdcBalance + _usdtBalance) * 1e18 / _3crvVirtualPrice * (BASIS_POINTS_GRANULARITY - depositMaxSlippageBasisPoints) / BASIS_POINTS_GRANULARITY;
+        require(_3crvBalanceFromDeposit >= _min3crvOut, "StableSwwapOperatorV1: 3pool deposit slippage too high");
+
+        // get the amount of tokens in the pool
+        (uint256 _3crvAmount, uint256 _feiAmount) = (
+            IStableSwap(pool).balances(_3crvIndex),
+            IStableSwap(pool).balances(_feiIndex)
+        );
+        // ... and the expected amount of 3crv in it after deposit
+        uint256 _3crvAmountAfter = _3crvAmount + _3crvBalanceAfter;
+
+        // get the usd value of 3crv in the pool
+        uint256 _3crvUsdValue = _3crvAmountAfter * _3crvVirtualPrice / 1e18;
+
+        // compute the number of FEI to deposit
+        uint256 _feiToDeposit = 0;
+        if (_3crvUsdValue > _feiAmount) {
+            _feiToDeposit = _3crvUsdValue - _feiAmount;
+        }
+
+        // deposit
+        if (_3crvBalanceAfter > 0 || _feiToDeposit > 0) {
+            // build parameters
+            uint256[2] memory _addLiquidityAmounts;
+            _addLiquidityAmounts[_feiIndex] = _feiToDeposit;
+            _addLiquidityAmounts[_3crvIndex] = _3crvBalanceAfter;
+
+            // approvals
+            IERC20(address(_fei)).approve(pool, _feiToDeposit);
+            // 3crv needs to reset allowance to 0, see CurveTokenV1.vy#L117 :
+            // assert _value == 0 or self.allowances[msg.sender][_spender] == 0
+            IERC20(_3crv).approve(pool, 0);
+            IERC20(_3crv).approve(pool, _3crvBalanceAfter);
+
+            // do deposit
+            uint256 _balanceBefore = IERC20(pool).balanceOf(address(this));
+            IStableSwap(pool).add_liquidity(_addLiquidityAmounts, 0);
+            uint256 _balanceAfter = IERC20(pool).balanceOf(address(this));
+
+            // emit event
+            uint256 _lpTotalSupply = IERC20(pool).totalSupply();
+            uint256 _3crvOut = (_3crvAmountAfter * (_balanceAfter - _balanceBefore)) / _lpTotalSupply;
+            uint256 _daiOut = IStableSwap(_3pool).calc_withdraw_one_coin(
+              _3crvOut, // LP tokens just deposited
+              0 // 3pool coin 0 = DAI
+            );
+            emit Deposit(msg.sender, _daiOut);
+        }
+
+        // burn remaining FEI
+        uint256 _feiBalance = _fei.balanceOf(address(this));
+        _fei.burn(_feiBalance);
```

Comment:
_burnFeiHeld() is a helper method in CoreRef. Not sure we want to do this though because we may want to keep the FEI we have since this contract isn't a minter

Reply:
I'll remove the FEI burn logics in `deposit()` and `withdraw()`. Since the contract can't mint new FEI, it makes sense to keep it now.

Reply:
btw, I didn't use _burnFeiHeld() to avoid a second call to fei() to optimize gas